### PR TITLE
Add routine to DataStore that performs sanity checks on observations

### PR DESCRIPTION
**Pull request by lmohrmann**
This simple routine performs some sanity checks on the events and IRF files of the observations in the DataStore.

### CODE REVIEW

**Comment by cdeil** [comment]
@lmohrmann - Thanks!

Ideally, I think it would be much better if we always implement checks by putting results in dicts like in
https://github.com/gammapy/gammapy/blob/master/gammapy/irf/psf_check.py
because then those can be logged, but could also be e.g. summarised in a table or more easily used in processing pipelines.

Another suggestion I have is that it would be nice if you add a test that exercises the new code at least once to make sure it's working (and keeps working as we refactor Gammapy). There's this CTA example
https://github.com/gammapy/gammapy-extra/tree/master/test_datasets/cta_1dc
and from HESS we have `datasets/hess-crab4-hd-hap-prod2` and `datasets/hess-crab4-pa` and if none of those works for some reason I could add a more recent one from HAP.

@lmohrmann - Let me know if you want to implement either of these suggestions here, or just merge as-is and leave those suggestions to the future.

File: gammapy/data/data_store.py [hunk 1]
```diff
@@ -24,6 +24,8 @@
     'ObservationList',
 ]
 
+format = "%(levelname)8s %(filename)s:%(lineno)s - %(funcName)s --- %(message)s"
+logging.basicConfig(level=logging.INFO, format=format)
```
Comment (cdeil) [inline gammapy/data/data_store.py]: Python libraries should never configure loggers. That makes it very hard for users to do so and confusing when `import gammapy.data` changes the global logging behaviour.

So if you keep the logging in `check_observations`, please remove those two lines here and instead put them in the script doing the processing (having it in cli scripts like `hap-data-fits-export` is OK).

File: gammapy/data/data_store.py [hunk 2]
```diff
@@ -284,6 +286,29 @@ def load_many(self, obs_ids, hdu_type=None, hdu_class=None):
 
         return things
 
+    def check_observations(self):
+        """Check all observations regarding their values in events, aeff, edisp, psf
```
Comment (cdeil) [inline gammapy/data/data_store.py]: Say in the docstring what the method does. I.e. if you keep as-is, say that it's logging the results.

**Comment by lmohrmann** [comment]
@cdeil I tried to implement your suggestions, see the new commit. The test in test_data_store.py fails if I use the CTA 1DC data or datasets/hess-crab4-pa because it cannot find the files at the expected locations.

File: gammapy/data/tests/test_data_store.py [hunk 3]
```diff
@@ -211,3 +211,10 @@ def test_make_mean_edisp(tmpdir):
     i = np.where(rmf.data.evaluate(e_reco=Energy(40, "TeV")) != 0)[0]
     i2 = np.where(rmf2.data.evaluate(e_reco=Energy(40, "TeV")) != 0)[0]
     assert_equal(i, i2)
+
+
+@requires_data('gammapy-extra')
```
Comment (cdeil) [inline gammapy/data/tests/test_data_store.py]: You have to add `@requires_dependency('yaml')` here: https://travis-ci.org/gammapy/gammapy/jobs/249970758#L1794

File: gammapy/data/data_store.py [hunk 4]
```diff
@@ -284,6 +284,43 @@ def load_many(self, obs_ids, hdu_type=None, hdu_class=None):
...
+        -------
+        OrderedDict : dictionary containing failure messages for all runs that fail a check.
+        """
+
+        results = OrderedDict()
+        nfail = 0
+
+        # Loop over all obs_ids in obs_table
+        for obs_id in self.obs_table['OBS_ID']:
+            messages = []
```
Comment (cdeil) [inline gammapy/data/data_store.py]: Everything in the for loop should be moved to a separate method. Would it fit better on the `obs` class?

File: gammapy/data/data_store.py [hunk 5]
```diff
@@ -284,6 +284,43 @@ def load_many(self, obs_ids, hdu_type=None, hdu_class=None):
...
+            if np.max(self.obs(obs_id).edisp.data.data) <= 0:
+                messages.append('maximum entry of energy dispersion table <= 0')
+            # Check that thresholds are meaningful for psf
+            if self.obs(obs_id).psf.energy_thresh_lo >= self.obs(obs_id).psf.energy_thresh_hi:
+                messages.append('LO_THRES >= HI_THRES in psf meta data')
+            if len(messages) > 0:
+                results[obs_id] = messages
+                nfail += 1
+
+        print('Checks failed for {:d} out of {:d} observations'.format(nfail, len(self.obs_table['OBS_ID'])))
```
Comment (cdeil) [inline gammapy/data/data_store.py]: If you `print` here, then it's very hard for callers to suppress (e.g. from pipelines).
I would suggest you leave this print to the caller, or if you want to do it from Gammapy, make a separate methods where you print the report.

**Comment by lmohrmann** [comment]
I tried to implement your suggestions again, see the latest commit. Let me know if this looks OK now.

**Comment by cdeil** [comment]
@lmohrmann - Thanks!

**Comment by cdeil** [comment]
@lmohrmann - I see at https://github.com/gammapy/gammapy/pull/1082/commits that you merged master into this feature branch:  b6495e48012c93881f95a082bcf1fe03e0d9fe47

This is not a problem in this case, we'll leave as-is.

But generally merge commits into feature branches means that overall the git commit history of the project is complex, and makes it harder to review / understand of when which changes happened. So for the future, please `rebase` on upstream `master` instead of merging it into your feature branch. We don't have git / developer docs for Gammapy, but it's the same as for Astropy (see http://astropy.readthedocs.io/en/latest/development/workflow/development_workflow.html#rebase-but-only-if-asked) or most other projects on Github.

**Pull request merged**